src/ImageSharp/Processing/Processors/Dithering/ErrorDiffuser.cs

```diff
@@ -82,45 +53,40 @@ public void Dither<TPixel>(ImageFrame<TPixel> image, TPixel source, TPixel trans
 
             // Calculate the error
             Vector4 error = source.ToVector4() - transformed.ToVector4();
-            this.DoDither(image, x, y, minX, minY, maxX, maxY, error);
+            this.DoDither(image, x, y, minX, maxX, maxY, error);
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private void DoDither<TPixel>(ImageFrame<TPixel> image, int x, int y, int minX, int minY, int maxX, int maxY, Vector4 error)
+        [MethodImpl(InliningOptions.ShortMethod)]
+        private void DoDither<TPixel>(ImageFrame<TPixel> image, int x, int y, int minX, int maxX, int maxY, Vector4 error)
             where TPixel : struct, IPixel<TPixel>
         {
+            int offset = this.startingOffset;
+            DenseMatrix<float> matrix = this.matrix;
+
             // Loop through and distribute the error amongst neighboring pixels.
-            for (int row = 0; row < this.matrixHeight; row++)
+            for (int row = 0, targetY = y + row; row < matrix.Rows && targetY < maxY; row++)
```

Comment:
Bug here. I'm not updating `targetY`

Reply:
Gutted about this. The bug was leasing to a 10x speedup in performance in the dither benchmarks but was leaving dithering areas too sparse only pushing errors to the left and not below..